packages/botonic-plugin-flow-builder/src/content-fields/flow-button.tsx

```diff
@@ -27,14 +27,19 @@ export class FlowButton extends ContentFieldsBase {
     const newButton = new FlowButton(cmsButton.id)
     newButton.text = this.getTextByLocale(locale, cmsButton.text)
     if (cmsButton.target) {
-      newButton.payload = cmsButton.target.id
+      const payloadFromBotActionNode = cmsApi.getPayloadFromBotActionNodeId(
+        cmsButton.target.id
+      )
+      const payload = payloadFromBotActionNode || cmsButton.target.id
```

Comment:
That's just what line 33 does.  The function getPayloadFromBotActionNodeId returns the payload defined in a BotAction if the id that you pass is a node is of type bot-action, if it is not of type undefined it returns undefined so that cmsButton.target.id will be executed.

Reply:
I agree with Adria

Reply:
refactored